src/connectors/refinement-list/__tests__/connectRefinementList-test.js

```diff
@@ -962,4 +976,131 @@ describe('connectRefinementList', () => {
       ]);
     });
   });
+
+  describe('routing', () => {
+    const getInitializedWidget = (config = {}) => {
+      const rendering = jest.fn();
+      const makeWidget = connectRefinementList(rendering);
+
+      const widget = makeWidget({
+        attributeName: 'facetAttribute',
+        ...config,
+      });
+
+      const initialConfig = widget.getConfiguration({}, {});
+      const helper = algoliasearchHelper(fakeClient, '', initialConfig);
+      helper.search = jest.fn();
+
+      widget.init({
+        helper,
+        state: helper.state,
+        createURL: () => '#',
+        onHistoryChange: () => {},
+      });
+
+      const { refine } = rendering.mock.calls[0][0];
+
+      return [widget, helper, refine];
+    };
+
+    describe('getWidgetState', () => {
+      test('should give back the object unmodified if the default value is selected', () => {
+        const [widget, helper] = getInitializedWidget();
+        const uiStateBefore = {};
+        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+          state: helper.state,
+          helper,
+        });
+        expect(uiStateAfter).toBe(uiStateBefore);
+      });
+
+      test('should add an entry equal to the refinement', () => {
+        const [widget, helper, refine] = getInitializedWidget();
+        refine('value');
+        const uiStateBefore = {};
+        const uiStateAfter = widget.getWidgetState(uiStateBefore, {
+          state: helper.state,
+          helper,
+        });
+        expect(uiStateAfter).toMatchSnapshot();
```

Comment:
You use Snapshot for those cases but I find them tedious in that case. Because by reading the tests file we have no clue of what it's expected.

Reply:
OK will had a more precise test.